omniqueue/src/backends/rabbitmq.rs

```diff
@@ -178,19 +193,49 @@ impl QueueConsumer for RabbitMqConsumer {
                 .clone()
                 .map(|l: Result<lapin::message::Delivery, lapin::Error>| {
                     let l = l.map_err(QueueError::generic)?;
-
-                    Ok(Delivery {
-                        decoders: self.registry.clone(),
-                        payload: Some(l.data),
-                        acker: Box::new(RabbitMqAcker {
-                            acker: Some(l.acker),
-                            requeue_on_nack: self.requeue_on_nack,
-                        }),
-                    })
+                    Ok(self.wrap_delivery(l))
                 });
 
         stream.next().await.ok_or(QueueError::NoData)?
     }
+
+    async fn receive_all(
+        &mut self,
+        max_messages: usize,
+        deadline: Duration,
+    ) -> Result<Vec<Delivery>, QueueError> {
+        let mut stream = self.consumer.clone();
+        let mut out = Vec::with_capacity(max_messages);
+
+        // FIXME: the real way to do this is to set the pre-fetch count on the channel (which happens much earlier).
```

Comment:
This is related to my other comment on the SQS implementation, but I don't think we should be "breaking" the default RabbitMQ behavior here.

If the `pre-fetch` count on the channel isn't >1, than just let `receive_all` only return a vector of one element. If the users of this library want `receive_all` to return more than one item, they can set the pre-fetch value when configuring their channel

Reply:
Maybe I can `stream.take(N)` or something to get a good behavior when the pre-fetch count is >1, but otherwise a problem might be recognizing the edges of the pre-fetch batches. A single `receive_all` call might end up crossing the boundary and actually result in multiple queue reads.

I'll add a public config value for a pre-fetch count and play around with it in the test suite to see how it works out.

Reply:
I exposed `consumer_prefetch_count` as an optional field on the rabbit config. It didn't have any impact on the tests, but it's a good knob to expose regardless.